src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -71,6 +70,17 @@ public partial class DynamoViewModel : ViewModelBase, IDynamoViewModel
         /// An observable collection of workspace view models which tracks the model
         /// </summary>
         private ObservableCollection<WorkspaceViewModel> workspaces = new ObservableCollection<WorkspaceViewModel>();
+
+        /// <summary>
+        /// Docked windows in sidebar
+        /// </summary>
+        internal Dictionary<string, NodeModel> DockedWindows { get; set; } = new Dictionary<string, NodeModel>();
+
+        /// <summary>
+        /// Docked node windows in sidebar
+        /// </summary>
+        internal Dictionary<string, ViewExtensionDisplayMode> NodeWindowStates { get; set; } = new Dictionary<string, ViewExtensionDisplayMode>();
```

Comment:
To me, looks like we only need the second dictionary here? It worries me if we store the NodeModel references in `DockedWindows` because if the Python node is removed in canvas, we still need to handle the disposal in addition?

Reply:
Yes addressed this in the above comment.